### ISSUE TITLE
Frequent crash in production. couldn't find libisar.so

### ISSUE THREAD

**Issue by KumarArab** [isar/isar#1080]
### Source: Firebase Crashlytics

So not able to replicate this while dev but on production, many users are facing frequent crashes right at the launch of the application.

### Code sample

```dart
Fatal Exception: java.lang.RuntimeException: Unable to get provider androidx.startup.InitializationProvider: androidx.startup.c: java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/in.fello.felloapp-DNOPCc4_kdV4Cxiu2kW8LQ==/base.apk"],nativeLibraryDirectories=[/data/app/in.fello.felloapp-DNOPCc4_kdV4Cxiu2kW8LQ==/lib/arm64, /system/lib64, /product/lib64]]] couldn't find "libisar.so"
       at android.app.ActivityThread.installProvider(ActivityThread.java:7194)
       at android.app.ActivityThread.installContentProviders(ActivityThread.java:6672)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6567)
       at android.app.ActivityThread.access$1400(ActivityThread.java:224)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1887)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:224)
       at android.app.ActivityThread.main(ActivityThread.java:7562)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

### Details

 - Platform: Android 12, 11, 10, 8, Xiaomi, Samsung, LG, etc.
 - Flutter version: 3.3.10
 - Isar version: 3.0.5

---

 - [x] I searched for similar issues already
 - [x] I filled the details section with the exact device model and version
 - [ ] I am able to provide a reproducible example

**Comment by Sadmansamee** [comment]
@KumarArab can you try with 3.0.6-dev.0?

**Comment by MaximilianFlechtner** [comment]
+1

**Comment by KumarArab** [comment]
> @KumarArab can you try with 3.0.6-dev.0?

Could you let me know if you are sure? you want me to try a dev version on a production app

**Comment by Sadmansamee** [comment]
yes

**Comment by bartosz-timecamp** [comment]
Did you manage to solve this issue? I am experiencing similar problem.

**Comment by zagalaga** [comment]
This happens to several users of mine, they all have one thing in common: Android 8 or 8.1. Devices Nexus 5x, motorola one vision. 
I am guessing those devices are looking for some architecture version of libisar.so and once they do not find it crash. This is a high severity issue because those users can't even use the app, it crashes on startup.

Using isar & isar_flutter_libs version 3.0.5

**Comment by JobiJoba** [comment]
Same issue but with a Samsung Galaxy A70 and Nexus 5 with Android 8.1 (isar 3.0.5)

**Comment by snipd-mikel** [comment]
We have also seen this issue in production: Android 8.1.0 and Nexus 5X

**Comment by ch-muhammad-adil** [comment]
Please tell me if someone is able to produce same issue at emulator.. I have tried Nexus 5 with API 27 but it works without problem.

**Comment by simc** [comment]
This is fixed in v3.1.0 which will be available soon on pub.dev.

**Comment by ch-muhammad-adil** [comment]
@simc Thanks a lot for making this fix. [Pull request created for update in documentation ](https://github.com/isar/isar/pull/1184)
